docs/proposals/service-design.md

```diff
@@ -0,0 +1,85 @@
+4. Only support HTTP communication in the first step
+5. All edge nodes are in the same sub-network and they have the network reachability.
+
+## Use cases
+<img src="../images/proposals/service-design.png">
+
+### Register the service located on cloud
+* CS0.1: User create service(type ClusterIP) in K8S API Server
+* CS0.2: EC get the service definition and create router rules with the edge servicebus as source and service clusterIP as target.
+* CS0.3: Send the service definition to edge
+
+### Register the service located on edge
+* CS1: User create service(headless service without ClusterIP) in K8S API Server
+* CS2, CS3: EC get the service definiton and update the service ClusterIP to itself. Then create router rule with the EC(router) address as the source and edge servicebus as target.
+* CS4: Send the service definition to edge
+
+### Service discovery on cloud
+* Application access service with the same mechanism no matter it located on cloud or edge
+
+### Service discovery on edge
+* Use iptables to redirect inbound and outbound trafic to the edgehub/servicebus. Then do serivce discovery in edgehub/servicebus by pick up the right POD from the service definition. The idea is similar with what istio is doing in the init_container: https://github.com/istio/istio/blob/master/tools/deb/istio-iptables.sh
+
+### Service communication from cloud to edge
+* ESD1: Application on cloud visit service with clusterIP and request goes to EC(refer to step CS2, CS3)
+* ESD2: Router find out the PODs in this service and pick one with some kind of load balance mechanism. Then send the request to the choosen edge with the router rule set in CS3.
+* ESD3: EdgeHub/ServiceBus get the request and redirect to the POD
+
+### Service communication between edges
+* ESR1: Application send the request to ServiceBus(refer to the "Service discovery on edge")
+* ESR2: ServiceBus pick one POD in this service with some kind of load balance mechanism. If the target is in the same node, just forward the request. If not, edgehub wrapper the request and send to the target.
+* ESR3: EdgeHub open the wrapped package and forward to the POD
+
+### Service communication from edge to cloud
+* ESU1: Application send the request to ServiceBus(refer to the "Service discovery on edge")
+* ESU2: ServiceBus pick one POD in this service with some kind of load balance mechanism. The target is on cloud, EdgeHub send the request to cloud.
+* ESU3: Router send the request to the corresponding POD on cloud following the rule created in CS0.2
+
+## Proposal Changes
+### Add servicebus on Edge to handle request proxy
+ServiceBus is in the same process with EdgeHub as a module
+
+### Router
+Add two endpoints definition:
```

Comment:
four endpoints definition?

Reply:
well, two is enough since each one can be both source and target